src/ngraph/op/fused/scatter_nd.cpp

```diff
@@ -0,0 +1,133 @@
+}
+
+void op::v0::ScatterND::pre_validate_and_infer_types()
+{
+    element::Type data_et = get_input_element_type(DATA);
+    element::Type indices_et = get_input_element_type(INDICES);
+    element::Type updates_et = get_input_element_type(UPDATES);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_et == element::i32 || indices_et == element::i64,
+                          "Indices element type must be i64 or i32.");
+
+    NODE_VALIDATION_CHECK(this,
+                          data_et == updates_et,
+                          "Updates element type must be the same as element type of data.");
+
+    const PartialShape& data_ps = get_input_partial_shape(DATA);
+    const PartialShape& indices_ps = get_input_partial_shape(INDICES);
+    const PartialShape& updates_ps = get_input_partial_shape(UPDATES);
+
+    if (data_ps.rank().is_static())
+    {
+        const size_t data_rank = static_cast<size_t>(data_ps.rank());
+        NODE_VALIDATION_CHECK(this, data_rank >= 1, "Data rank is expected to be at least 1.");
+    }
+
+    if (indices_ps.rank().is_static())
+    {
+        const size_t indices_rank = static_cast<size_t>(indices_ps.rank());
+
+        NODE_VALIDATION_CHECK(
+            this, indices_rank >= 1, "Indices rank is expected to be at least 1.");
+    }
+
+    if (indices_ps.rank().is_static() && data_ps.rank().is_static())
+    {
+        const size_t indices_rank = static_cast<size_t>(indices_ps.rank());
+        const size_t last_dim_pos = indices_rank - 1;
+        const Dimension indices_last_dim = indices_ps[last_dim_pos];
+        if (indices_last_dim.is_static())
+        {
+            const size_t indices_last_dim_value = static_cast<size_t>(indices_last_dim);
+            const size_t data_rank = static_cast<size_t>(data_ps.rank());
+            NODE_VALIDATION_CHECK(this,
+                                  indices_last_dim_value <= data_rank,
+                                  "Last dimension of indices can be at most the rank of data.");
+
+            if (updates_ps.rank().is_static())
+            {
+                const size_t expected_updates_rank =
+                    data_rank + indices_rank - indices_last_dim_value - 1;
+
+                NODE_VALIDATION_CHECK(
+                    this,
+                    static_cast<size_t>(updates_ps.rank()) == expected_updates_rank,
+                    "Updates rank is expected to be equal data_rank + indices_rank - "
+                    "indices_shape[-1] - 1.");
+            }
+        }
+    }
+
+    set_output_type(0, data_et, data_ps);
+}
+
+NodeVector op::ScatterND::decompose_op() const
+{
+    const auto data = input_value(DATA);
+    const auto indices = input_value(INDICES);
+    const auto updates = input_value(UPDATES);
+
+    const Shape& data_shape = get_input_shape(DATA);
+    const Shape& updates_shape = get_input_shape(UPDATES);
+
+    element::Type data_et = get_input_element_type(DATA);
```

Comment:
Will be deprecated. Please replace with `node->input(i).get_element_type()`

Reply:
Done